routemaster/state_machine/actions.py

```diff
@@ -88,6 +94,12 @@ def _process_action_with_metadata(
     return True
 
 
+def _calculate_idempotency_token(label: LabelRef, history_id: int):
```

Comment:
Could annotate the return type here.

Reply:
A docstring would also be nice (explaining a little of the role of idempotency tokens and their semantics).

Reply:
I've generally tidied this up a lot in 3e7b914b5b2078ccc710e251a560b59b1c3cee15